pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -302,8 +302,13 @@ Map<String, List<Operation>> computeOperations(SegmentDirectory.Reader segmentRe
         }
 
         // Note that RAW columns cannot be sorted.
+        // TODO: revisit the the possibility of converting noDictionary -> Dictionary when the column is sorted.
         ColumnMetadata existingColMetadata = _segmentDirectory.getSegmentMetadata().getColumnMetadataFor(column);
-        Preconditions.checkState(!existingColMetadata.isSorted(), "Raw column=" + column + " cannot be sorted.");
+        if (existingColMetadata.isSorted()) {
+          LOGGER.warn("Raw column={} cannot be sorted and the dictionary won't be created. Falling back to use "
+              + "noDictionaryColumn..", column);
+          continue;
+        }
```

Comment:
I think we can remove the whole check. This check itself doesn't provide extra value

Reply:
We don't want to skip this column though

Reply:
done